cabal-testsuite/PackageTests/ConstraintSource/SourceRepositoryPackage/cabal.out

```diff
@@ -0,0 +1,16 @@
+# cabal v2-update
+Downloading the latest package list from test-local-repo
+# cabal v2-build
+Configuration is affected by the following files:
+- cabal.project
+Resolving dependencies...
+Error: [Cabal-7107]
+Could not resolve dependencies:
+[__0] next goal: my-lib (user goal)
+[__0] rejecting: my-lib; 2.0, 1.0
+      (constraint from cabal.project requires ==0.9)
```

Comment:
I have not checked the code, but this output looks very good. I don't know why the newline though instead of printing it in the same line.

Reply:
I quite like the new line fwiw. Of course, it’s impossible to find a consensus around such things…

Reply:
This very nice output formatting is thanks to @philderbeast in #9578 😁

Reply:
> I don't know why the newline though instead of printing it in the same line.

The rejecting message can be quite long and looks alright with a flow layout but showing project and config constraints with their "imported by" parent chain didn't so I went with an indented layout for those, starting on a new line:

 https://github.com/haskell/cabal/blob/c3a9dd7722335900619d64ad19be79ec5b7cf402/cabal-testsuite/PackageTests/ConditionalAndImport/cabal.out#L242-L253